src/App.tsx

```diff
@@ -14,11 +14,140 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  isReversed: boolean,
+  isSortedBy: string,
```

Comment:
isSortedBy sounds like a boolean value (true/false)
also, create SortBy enum
```suggestion
  sortBy: SortByEnum,
```